Microsoft.Toolkit.Uwp/Helpers/ObjectStorage/BaseObjectStorageHelper.cs

```diff
@@ -115,7 +126,17 @@ public T Read<T>(string compositeKey, string key, T @default = default(T))
         /// <param name="value">Object to save</param>
         public void Save<T>(string key, T value)
         {
-            Settings.Values[key] = JsonConvert.SerializeObject(value);
+            var type = typeof(T);
+            var typeInfo = type.GetTypeInfo();
+
+            if (typeInfo.IsPrimitive || type == typeof(string))
```

Comment:
I am curious. What types should support the "simple" version of the Read/Save method.  This check does not include `Decimal`, `DateTime`, `TimeSpan` or `Guid`.

Reply:
https://msdn.microsoft.com/en-us/library/system.type.isprimitive(v=vs.110).aspx

The primitive types are Boolean, Byte, SByte, Int16, UInt16, Int32, UInt32, Int64, UInt64, IntPtr, UIntPtr, Char, Double, and Single. All others fallback to previously used manner

Reply:
Decimal, DateTime, TimeSpan and Guid are not simple types, while they may be structs, they are far from simple objects

Reply:
Is it the expected behavior? Like for Decimal types. Anyway, I suppose it should equally work using both methods, right?

Reply:
Sorry @skendrot, just saw your message. Thanks for your explanation.